components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/TokenValidationHandler.java

```diff
@@ -682,17 +679,20 @@ private static boolean isOmitUsernameInIntrospectionRespForAppTokens(AccessToken
                                                                          String tenantDomain)
             throws IdentityOAuth2Exception {
 
-        OAuthAdminServiceImpl oAuthAdminService = OAuthComponentServiceHolder.getInstance().getoAuthAdminService();
-        boolean omitUsernameInIntrospectionRespForAppTokens;
         try {
-            OAuthConsumerAppDTO oAuthApp = oAuthAdminService.getOAuthApplicationData(accessTokenDO.getConsumerKey(),
+            int appResidentTenantId = accessTokenDO.getAppResidentTenantId();
+            if (appResidentTenantId != MultitenantConstants.INVALID_TENANT_ID) {
+                tenantDomain = IdentityTenantUtil.getTenantDomain(appResidentTenantId);
+            }
+            OAuthAppDO oAuthAppDO = OAuth2Util.getAppInformationByClientId(accessTokenDO.getConsumerKey(),
                     tenantDomain);
-            omitUsernameInIntrospectionRespForAppTokens = oAuthApp.isOmitUsernameInIntrospectionRespForAppTokens();
+
+            return Boolean.TRUE.equals(oAuthAppDO.isOmitUsernameInIntrospectionRespForAppTokens());
```

Comment:
cant we just return the `oAuthAppDO.isOmitUsernameInIntrospectionRespForAppTokens()`?

Reply:
We can. But to be safe used this.

Reply:
oAuthAppDO.isOmitUsernameInIntrospectionRespForAppTokens() never be null right? then its ok